esmf_regrid/experimental/unstructured_scheme.py

```diff
@@ -101,8 +97,34 @@ def _create_cube(data, src_cube, mesh_dim, grid_x, grid_y):
 
     new_cube.metadata = copy.deepcopy(src_cube.metadata)
 
-    for coord in src_cube.coords(dimensions=()):
-        new_cube.add_aux_coord(coord.copy())
+    # TODO: Handle derived coordinates. The following code is taken from
+    #  iris, the parts dealing with derived coordinates have been
+    #  commented out for the time being.
+    # coord_mapping = {}
+
+    def copy_coords(src_coords, add_method):
+        for coord in src_coords:
+            dims = src_cube.coord_dims(coord)
+            if hasattr(coord, "mesh") or mesh_dim in dims:
+                continue
+            dims = [dim if dim < mesh_dim else dim + 1 for dim in dims]
+            result_coord = coord.copy()
+            add_method(result_coord, dims)
```

Comment:
I *think* I get this, but slightly confusing as we're not returning anything from this function.  Is this because `add_method` is always going to be a method on a cube to which we are now appending these new coordinates, so the state change happens on the `add_method` owner?